From 789b29e1a272ebdd97aae7abcaa221308dbda0f5 Mon Sep 17 00:00:00 2001
From: Jocelyn Falempe <jfalempe@redhat.com>
Date: Fri, 21 Aug 2026 09:38:43 +0200
Subject: [PATCH] seat: Fix use after free in kmscon_seat_remove_video()

As the display is removed from the list during traversal, use the safe
variant

Signed-off-by: Jocelyn Falempe <jfalempe@redhat.com>
---
 src/seat.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/seat.c b/src/seat.c
index 299620a34600..ea0b9b088421 100644
--- a/src/seat.c
+++ b/src/seat.c
@@ -1015,7 +1015,7 @@ static void kmscon_seat_remove_video(struct kmscon_seat *seat, void *data)
 {
 	struct kmscon_video *vid = data;
 	struct kmscon_display *d;
-	struct shl_dlist *iter;
+	struct shl_dlist *iter, *tmp;
 
 	if (!seat || !vid)
 		return;
@@ -1026,7 +1026,7 @@ static void kmscon_seat_remove_video(struct kmscon_seat *seat, void *data)
 	shl_dlist_unlink(&vid->list);
 
 	if (vid->video) {
-		shl_dlist_for_each(iter, &seat->displays)
+		shl_dlist_for_each_safe(iter, tmp, &seat->displays)
 		{
 			d = shl_dlist_entry(iter, struct kmscon_display, list);
 			if (display_video(d->disp) == vid->video)
-- 
2.55.0

